PWGJE/TableProducer/emcalCellTrackMatcher.cxx

```diff
@@ -0,0 +1,221 @@
+    int nBCsProcessed = 0;
+    int nCellsProcessed = 0;
+    const int trackDefValue = -900;
+    for (const auto& bc : bcs) {
+      LOG(debug) << "Next BC";
+
+      // Get the collisions matched to the BC using foundBCId of the collision
+      auto collisionsInFoundBC = collisions.sliceBy(collisionsPerFoundBC, bc.globalIndex());
+      auto cellsInBC = cells.sliceBy(cellsPerFoundBC, bc.globalIndex());
+
+      if (!cellsInBC.size()) {
+        LOG(debug) << "No cells found for BC";
+        continue;
+      }
+      // We need bcs which only have a single collision matched to them,
+      // so that we can get the tracks from that collision
+      if (collisionsInFoundBC.size() == 1) {
+        // For the matching of a track to a cell we can only get the geomtrical cell ID, however,
+        // the cells inside the cell table do not store this info. So to later easily check if a cell has a hit
+        // and translate from geometrical ID to table cellID we will create an unordered_map
+        // Create the geometrical-to-logical cell map for the current BC and fill it
+        std::unordered_map<int, int32_t> mapGeoToLogical;                  // geoId -> logicalId
+        std::unordered_map<int32_t, std::vector<int32_t>> mapCellToTracks; // cell -> vector of trackIDs
+        mapGeoToLogical.reserve(cellsInBC.size());
+        mapCellToTracks.reserve(cellsInBC.size());
+        for (const auto& cell : cellsInBC) {
+          mapGeoToLogical[cell.cellNumber()] = cell.globalIndex(); // Mapping geometrical ID to table cell ID
+        }
+        LOG(detail) << "Number of cells for BC (CF): " << cellsInBC.size();
+        nCellsProcessed += cellsInBC.size();
+        // dummy loop to get the first collision
+        for (const auto& col : collisionsInFoundBC) {
+          if (previousCollisionId > col.globalIndex()) {
+            mHistManager.fill(HIST("hBCMatchErrors"), 1);
+            continue;
+          }
+          previousCollisionId = col.globalIndex();
+          if (col.foundBCId() == bc.globalIndex()) {
+            mHistManager.fill(HIST("hBCMatchErrors"), 0); // CollisionID ordered and foundBC matches -> Fill as healthy
+
+            auto groupedTracks = tracks.sliceBy(perCollision, col.globalIndex());
+            for (const auto& track : groupedTracks) {
+              if (!track.isGlobalTrack()) { // only global tracks
+                continue;
+              }
+              // Tracks that do not point to the EMCal/DCal/PHOS get default values of -999
+              // This way we can cut out tracks that do not point to the EMCal+DCal
+              if (track.trackEtaEmcal() < trackDefValue || track.trackPhiEmcal() < trackDefValue) {
+                continue;
+              }
+              if (trackMinPt > 0 && track.pt() < trackMinPt) {
+                continue;
+              }
+              if ((track.dcaZ() / std::sqrt(track.cZZ()) > trackDCAzSigma) || (track.dcaXY() / std::sqrt(track.cYY()) > trackDCAxySigma)) {
```

Comment:
why is this used? I think we do not apply this for normal track matching

Reply:
In AliPhysics for the Cell Track Matcher we had the option to only match "primary" tracks. 
This was done by cutting on DCA_xy of 5 cm.
I was unsure how we want to do this, so I just implemented a version where one can cut on the DCA_z and DCA_xy but with the sigma instead of the values to account for the tracking precision.
This is just the way I have done it now and it is open for discussion how we want to have this done.